opencart 1.5.6.4/upload/catalog/controller/module/smaily_for_opencart.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * Front end newsletter widget.
+ */
+class ControllerModuleSmailyForOpencart extends Controller {
+
+    public function index() {
+        // Load Smaily settings.
+        $this->load->model('setting/setting');
+        $settings = $this->model_setting_setting->getSetting('smaily');
```

Comment:
I tried loading smailyforopencart/admin and using getSettingValue, but OpenCart can only load models that are in the same folder. The core files for loading use the current directory (catalog vs admin in this case) and I didn't want to edit those.

Reply:
OK.

But if you want to make your life easier, you could also use the same pattern as OpenCart 1.5 does - it has two different `settings/settings` models, one is under `admin` and the other under `catalog` (latter contains only the getter method of settings).

Reply:
> But if you want to make your life easier, you could also use the same pattern as OpenCart 1.5 does - it has two different `settings/settings` models, one is under `admin` and the other under `catalog` (latter contains only the getter method of settings).

Not sure I completely understand what you mean by this. I am using the catalog `settings/settings` model in this case?

Reply:
What I meant by it is, you could create another model under `catalog/model/smailyforopencart` (like you have under `admin/model/smailyforopencart`) that would implement the same `getSettingValue` method. It would make getting specific settings easier and your code a little bit cleaner, but using `catalog/model/settings/settings` is fine also.